packages/features/users/components/UserTable/BulkActions/EventList.tsx

```diff
@@ -0,0 +1,193 @@
+
+import classNames from "@calcom/lib/classNames";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { SchedulingType } from "@calcom/prisma/enums";
+import { trpc } from "@calcom/trpc";
+import type { RouterOutputs } from "@calcom/trpc/react";
+import {
+  Button,
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+  Popover,
+  showToast,
+  PopoverContent,
+  PopoverTrigger,
+} from "@calcom/ui";
+
+import type { User } from "../UserListTable";
+
+interface Props {
+  table: Table<User>;
+  teams: RouterOutputs["viewer"]["organizations"]["getTeams"] | undefined;
+}
+
+export function EventListBulkAction({ table, teams }: Props) {
+  const { t } = useLocale();
+  const utils = trpc.useUtils();
+  const teamIds = teams?.map((team) => team.id);
+  const { data } = trpc.viewer.eventTypes.getByViewer.useQuery({
+    filters: { teamIds, schedulingType: SchedulingType.ROUND_ROBIN },
+  });
+  const mutation = trpc.viewer.organizations.bulkAddToEventType.useMutation({
+    onError: (error) => {
+      showToast(error.message, "error");
+    },
+    onSuccess: (res) => {
+      showToast(`${res.count} Users added to ${Array.from(selectedEvents).length} events`, "success");
+
+      utils.viewer.organizations.listMembers.invalidate();
+      utils.viewer.eventTypes.invalidate();
+
+      // Clear the selected values
+      setSelectedEvents(new Set());
+      setSelectedTeams(new Set());
+      table.toggleAllRowsSelected(false);
+    },
+  });
+  const [selectedEvents, setSelectedEvents] = useState<Set<number>>(new Set());
+  const [selectedTeams, setSelectedTeams] = useState<Set<number>>(new Set());
+  const eventTypeGroups = data?.eventTypeGroups;
+
+  // Add value array to the set
+  const addValue = (set: Set<number>, setSet: Dispatch<SetStateAction<Set<number>>>, value: number[]) => {
+    const updatedSet = new Set(set);
+    value.forEach((v) => updatedSet.add(v));
+    setSet(updatedSet);
+  };
+
+  // Remove value array from the set
+  const removeValue = (set: Set<number>, setSet: Dispatch<SetStateAction<Set<number>>>, value: number[]) => {
+    const updatedSet = new Set(set);
+    value.forEach((v) => updatedSet.delete(v));
+    setSet(updatedSet);
+  };
+
+  return (
+    <>
+      <Popover>
+        <PopoverTrigger asChild>
+          <Button StartIcon="users">{t("add_to_event_type")}</Button>
+        </PopoverTrigger>
+        <PopoverContent className="w-[200px] p-0 shadow-md" align="start" sideOffset={12}>
+          <Command>
+            <CommandInput placeholder={t("search")} />
+            <CommandList>
+              <CommandEmpty>No results found.</CommandEmpty>
+              <CommandGroup>
+                {eventTypeGroups &&
+                  eventTypeGroups.map((team) => {
+                    const events = team.eventTypes;
+                    const ids = events.map((event) => event.id);
+                    const teamId = team.teamId || 0;
```

Comment:
I would remove `|| 0` and render `ListItem` only if `teamId` is defined - code should not render anything if the `teamId` is not given because if it did, then something is broken, and that data would be sent to backend - the logic should only work if `teamId` is defined.